src/server/api/v1/user.ts

```diff
@@ -0,0 +1,266 @@
+import express from 'express'
+import * as UserApi from '../../../userApi'
+import * as CurrencyApi from '../../../currencyApi'
+import Transaction from '../../../db/models/transaction'
+import cookieParser from 'cookie-parser'
+
+const router = express.Router()
+
+/** Parsers */
+router.use(cookieParser()) // for parsing cookie
+router.use(express.json()) // for parsing application/json
+
+/**
+ * Checa se você está logado
+ */
+router.use(async (req, res, next) => {
+	try {
+		if (!req.cookies.sessionId) throw 'Cookie Not Found'
+		req.user = await UserApi.findUser.byCookie(req.cookies.sessionId)
+		next()
+	} catch(err) {
+		res.status(401).send({
+			error: 'NotAuthorized',
+			message: 'A valid cookie \'sessionId\' needs to be informed to perform this operation'
+		})
+	}
+})
+
+/**
+ * Retorna todas as contas do usuario
+ */
+router.get('/accounts', (req, res) => {
+	const account: object = {}
+	for (const currency of CurrencyApi.currencies) {
+		account[currency] = req.user?.getAccounts(currency)
+	}
+	res.send(account)
+})
+
+/**
+ * Retorna todos os saldos do usuario
+ */
+router.get('/balances', (req, res) => {
+	const balance: object = {}
+	for (const currency of CurrencyApi.currencies) {
+		balance[currency] = req.user?.getBalance(currency, true)
+	}
+	res.send(balance)
+})
+
+router.get('/info', async (_req, res) => {
+	res.send({ info: 'info' })
+})
+
+/**
+ * Retorna uma transação especifica do usuario
+ */
+router.get('/transactions/:opid', async (req, res) => {
+	try {
+		const transactions = await Transaction.findById(req.params.opid)
```

Comment:
O que acontece se o parâmetro opid for inválido ou nulo?

Reply:
Se ele for invalido ele não vai encontrar a transação o que vai cair no if e disparar um not found,
agora quando você diz nulo, você quer dizer quando ele não envia o opid?
Se for isso, então não vai cair ai e sim no '/transactions'

Reply:
Hmmmm faz sentido

E se a string do opid for de um tamanho diferente do tamanho que um objectId tem (maior ou menor)? Não dá nenhum erro?

Reply:
Se a string for invalida seja por tamanho diferente ou algo problema, transactions vai retornar um undefined o que vai ser barrado no if dando throw 'notFound'

Reply:
Hmmm certo. Se vc **testou** e realmente é isso então ok